src/bridge/MatrixBanSync.ts

```diff
@@ -0,0 +1,133 @@
+/**
+ * Synchronises Matrix `m.policy.rule` events with the bridge to filter specific
+ * users from using the service.
+ */
+
+import { Intent, MatrixUser, WeakStateEvent } from "matrix-appservice-bridge";
+import { MatrixGlob } from "matrix-bot-sdk";
+import { getLogger } from "../logging";
+
+const log = getLogger("MatrixBanSync");
+export interface MatrixBanSyncConfig {
+    rooms: string[];
+}
+
+enum BanEntityType {
+    Server = "m.policy.rule.server",
+    User = "m.policy.rule.user"
+}
+
+interface BanEntity {
+    matcher: MatrixGlob;
+    entityType: BanEntityType;
+    reason: string;
+}
+
+interface MPolicyContent {
+    entity: string;
+    reason: string;
+    reccomendation: "m.ban";
+}
+
+function eventTypeToBanEntityType(eventType: string): BanEntityType|null {
+    switch (eventType) {
+        case "m.policy.rule.user":
+            return BanEntityType.User;
+        case "m.policy.rule.server":
+            return BanEntityType.Server
+        default:
+            return null;
+    }
+}
+
+export class MatrixBanSync {
+    private bannedEntites = new Map<string, BanEntity>();
+    private interestingRooms = new Set<string>();
```

Comment:
I'm not too sure about that name :P Then again, it's a private API (except for when the tests happily ignore that fact), so I don't really mind – in the context of this class it's obvious what it's for. Though maybe `eventSources` would be more representative of its uses?